server/remote_asset/fetch_server/fetch_server.go

```diff
@@ -198,7 +198,24 @@ func mirrorToCache(ctx context.Context, bsClient bspb.ByteStreamClient, remoteIn
 	if rsp.StatusCode < 200 || rsp.StatusCode >= 400 {
 		return nil, status.UnavailableErrorf("failed to fetch %q: HTTP %s", uri, err)
 	}
-	// Download to a temp file to avoid running out of memory.
+
+	// If we know what the SHA256 should be and the content length is known,
+	// then we know the full digest, and can pipe directly from the HTTP
+	// response to cache.
+	if expectedSHA256 != "" && rsp.ContentLength >= 0 {
+		d := &repb.Digest{Hash: expectedSHA256, SizeBytes: rsp.ContentLength}
+		rn := digest.NewResourceName(d, remoteInstanceName)
```

Comment:
Should be NewCASResourceName

Reply:
Hm why is the CacheType needed in this case? The ByteStream API (used here) doesn't have a notion of CacheType -- in ResourceName.UploadString() we don't use the CacheType.

If we want to have CacheType defined everywhere for consistency reasons, should we get rid of NewCASResourceName() and have NewResourceName() default to CacheType.CAS (since AC is more of a special case?)

Reply:
Ah you're right - my bad. Doesn't like the cachetype is needed here